### PR TITLE
Created a mk3 version.

### PR DESCRIPTION
**Pull request by dhowlett99**
Hello I have created a MK3 version of this code, tested with a LaunchPad Mini MK3

I have responed to your comments,  do you think we could merge this is now ?

### CODE REVIEW

File: mk3/launchpad.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+			if info.IsInputAvailable {
+				in = i
+			}
+			if info.IsOutputAvailable {
+				out = i
+			}
+		}
+	}
+	if in == -1 || out == -1 {
+		err = errors.New("launchpad: no launchpad is connected")
```
Comment (rakyll) [inline mk3/launchpad.go:165]: return errors.New("launchpad: no launchpad is connected")

File: mk3/launchpad.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	} else {
```
Comment (rakyll) [inline mk3/launchpad.go:166]: Remove the else.

File: mk3/launchpad.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	"time"
+
+	"github.com/rakyll/portmidi"
+)
+
+// Launchpad represents a device with an input and output MIDI stream.
+type Launchpad struct {
+	inputStream  *portmidi.Stream
+	outputStream *portmidi.Stream
+	Block        bool
```
Comment (rakyll) [inline mk3/launchpad.go:32]: Unexport this, BlockKeys is the setter.

File: mk3/launchpad.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+				}
+			} else {
+				pad.Read() // Throw away whilst we're blocked.
+			}
+		}
+	}(l, ch)
+	return ch
+}
+
+func (l *Launchpad) BlockKeys(block bool) {
```
Comment (rakyll) [inline mk3/launchpad.go:87]: Do we expect this to be safe concurrent use?

**Review by rakyll — COMMENTED** [review]
This looks great, I added a few minor suggestions.

File: mk3/launchpad.go (hunk 1 above)
Comment (dhowlett99) [reply mk3/launchpad.go:152]: Not sure what you mean, the error is already there.

File: mk3/launchpad.go (hunk 2 above)
Comment (dhowlett99) [reply mk3/launchpad.go:153]: comes from your original code. i think you need the else to be there to set the correct values if there is indeed a launchpad connnected.

File: mk3/launchpad.go (hunk 4 above)
Comment (dhowlett99) [reply mk3/launchpad.go]: removed the block keys feature as this was not my problem I was trying to solve.

File: mk3/launchpad.go (hunk 3 above)
Comment (dhowlett99) [reply mk3/launchpad.go]: removed this all together

File: mk3/launchpad.go (hunk 1 above)
Comment (dhowlett99) [reply mk3/launchpad.go:152]: Looking at this again,  the code is a strangley structured but you need the else case to set the three input variables as the returned ```input portmidi.DeviceID, output portmidi.DeviceID, err error``` and only setting the error if needed.

Comment (rakyll) [reply mk3/launchpad.go:152]: I'll send a follow up to fix what I mean, no worries.

**Review by rakyll — APPROVED** [review]

**Comment by rakyll** [comment]
Thanks much!

**Pull request merged**